kodilla-good-patterns/src/main/java/flightSearcher/FindFlight.java

```diff
@@ -0,0 +1,39 @@
+package flightSearcher;
+
+import javafx.util.Pair;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FindFlight {
+
+    public Flight flight;
+    CreateFlightList createFlightList = new CreateFlightList();
+
+    public FindFlight(Flight flight) {
+        this.flight = flight;
+    }
+
+    public List<Flight> findFlightTo() {
+        List<Flight> flights = createFlightList.getAllFlights();
+
+        return flights.stream()
+                .filter(flight1 -> flight1.getArrival().equals(flight.getArrival()))
+                .collect(Collectors.toList());
+    }
+
+    public List<Flight> findFlightFrom() {
+        List<Flight> flights = createFlightList.getAllFlights();
+
+        return flights.stream()
+                .filter(flight1 -> flight1.getDeparture().equals(flight.getDeparture()))
+                .collect(Collectors.toList());
+    }
+
+    public  List<Pair<Flight, Flight>> findConnectingFlight() {
+        List<Pair<Flight, Flight>> pairList = createFlightList.getAllPairs();
```

Comment:
tutaj masz streamowac liste getAllFlights()

Reply:
To rzeczywiście ułatwia, bo przy dodaniu nowego lotu od razu jest on uwzględniany, jednak mam problem z ułożeniem warunku... Cały czas szuka mi lotów bezpośrednich